tests/test_commands.py

```diff
@@ -0,0 +1,188 @@
+import os
+import signal
+import shutil
+import subprocess
+import sys
+import time
+
+import pytest
+import requests
+import sh
+
+
+FIXTURES = 'tests/fixtures'
+WEBSITE = os.path.join(FIXTURES, 'website.py')
+BUILDDIR = os.path.join(FIXTURES, '_build')
+INDEX = os.path.join(BUILDDIR, 'index.html')
+CNAME = os.path.join(BUILDDIR, 'CNAME')
+
+
+def parametrized_fixture(*args, **kwargs):
+    @pytest.fixture(scope='module', params=args + tuple(kwargs.keys()))
+    def fixture(request):
+        if request.param in kwargs:
+            return kwargs[request.param]
+        return request.param
+    return fixture
+
+
+class Cname(str):
+    '''
+    Helper class that is str, but has different truthfulness
+    '''
+    def __bool__(self):
+        return self == ' --cname'
+
+
+cname = parametrized_fixture(cname=Cname(' --cname'),
+                             no_cname=Cname(' --no-cname'))
+serve_command = parametrized_fixture('serve', freeze_serve='freeze --serve')
+port = parametrized_fixture(8001, 8080)
+domain = parametrized_fixture('foo.bar', 'spam.eggs')
+protocol = parametrized_fixture('http', 'https')
+
+
+class ElsaRunner:
+    '''
+    Class for elsa fixture enabling multiple runs and exit status check
+    '''
+    def __init__(self):
+        self.runs = []
+
+    def run(self, command):
+        print('COMMAND: python website.py', command)
+        proc = subprocess.Popen([sys.executable, WEBSITE] + command.split())
```

Comment:
What do you mean by "command line"? I think there are two options:
* what the program actually gets – `argv`, which is a list (or C array of strings)
* what you type into /bin/sh, which is a string that can contain shell metacharacters (such as spaces), and gets converted to `argv` by a rather intractable process

> Note – `subprocess` supports both, but the second is generally discouraged: it's there mainly for ease of porting from the [many functions it deprecated](https://docs.python.org/3/library/subprocess.html#replacing-os-popen-os-popen2-os-popen3), which mainly take strings because they wrap [C functions](https://linux.die.net/man/3/system), and C arrays are cumbersome to use.

Anyway, a string of space-separated words is neither of these, and it's setting you up to needing additional hacks in the future.